### PR TITLE
Added timeutils package

### PR DESCRIPTION
**Pull request by Sundin**
The timeutils package contains some useful helper methods for working with time and timestamps.

### CODE REVIEW

File: timeutils/month.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+const earliestYear = 1801
+const latestYear = 2999
+
+// GetPeriodsStartAndEndUTC returns the start and end timestamps (in milliseconds) given two months for location UTC (timezone)
+func GetPeriodsStartAndEndUTC(firstyyyymm string, lastyyyymm string) (start int64, end int64, err error) {
+	return getPeriodsStartAndEnd(firstyyyymm, lastyyyymm, *time.UTC)
+}
+
+// getPeriodsStartAndEnd returns the period start and end timestamps (in milliseconds) given two months and the location (timezone)
+func getPeriodsStartAndEnd(firstyyyymm string, lastyyyymm string, location time.Location) (start int64, end int64, err error) {
```
Comment (stroem) [inline timeutils/month.go]: unnecessary function, this code could be executed directly under `GetPeriodsStartAndEndUTC`
It is always time.UTC. Or should this function be exported?

File: timeutils/milli.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+// AssertMilliseconds ensures that a given timestamp is in milliseconds.
+// If not, a timestamp converted into milliseconds as well as an error will be returned.
+func AssertMilliseconds(timestamp int64) (timestampMilliseconds int64, err error) {
+	var timestampYear3000Milliseconds int64 = 32503680000000
+	for timestamp > timestampYear3000Milliseconds {
+		// Make sure timestamp is not in nanosecond or microsecond format
+		timestamp /= 1000
+		err = errors.New("got timestamp that was not milliseconds, had to convert")
```
[the remark this answers is not in the document]
Comment (Sundin) [reply timeutils/milli.go]: not in practice, the timestamp will either be too large, too small but positive, or below 0 - not all three

[the remark this answers is not in the document]
Comment (Sundin) [reply timeutils/milli.go]: Comment for `AssertSeconds` is missing

File: timeutils/month.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		end = 0
+		err = fmt.Errorf("start %s may not be after end %s", firstyyyymm, lastyyyymm)
+	}
+
+	return
+}
+
+// getMonthStartAndEnd returns timstamps (in milliseconds) for the month's start and end for the given location
+func getMonthStartAndEnd(yyyymm string, location time.Location) (start int64, end int64, err error) {
+	syyyy, smm, err := validFormat(yyyymm)
```
[the remark this answers is not in the document]
Comment (Sundin) [reply timeutils/month.go]: that snippet gives month = january for all inputs

File: timeutils/milli.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sundin) [reply timeutils/milli.go]: rewrote it in a cleaner way

File: timeutils/seconds.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package timeutils
+
+import (
+	"errors"
+)
+
+// AssertSeconds ensures that a given timestamp is in seconds.
+// If not, a timestamp converted into seconds as well as an error will be returned.
+func AssertSeconds(timestamp int64) (timestampSeconds int64, err error) {
```
[the remark this answers is not in the document]
Comment (Sundin) [reply timeutils/seconds.go]: Resulted in int64 overflow when working with nanoseconds

File: timeutils/milli.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
+package timeutils
+
+import (
+	"errors"
+	"time"
+)
+
+// Milliseconds returns milliseconds for time now
+func Milliseconds() int64 {
```
[the remark this answers is not in the document]
Comment (Sundin) [reply timeutils/milli.go]: Good!

**Pull request merged**